lib/buildingsync/model_articulation/spatial_element.rb

```diff
@@ -281,18 +281,18 @@ def add_element_in_xml_file(building_element, ns, field_name, field_value)
     end
 
     def write_parameters_to_xml_for_spatial_element(ns, xml_element)
-      add_element_in_xml_file(xml_element, ns, 'TotalFloorArea', @total_floor_area)
-      add_element_in_xml_file(xml_element, ns, 'BuildingType', @bldg_type)
-      add_element_in_xml_file(xml_element, ns, 'SystemType', @system_type)
-      add_element_in_xml_file(xml_element, ns, 'BarDivisionMethod', @bar_division_method)
-      add_element_in_xml_file(xml_element, ns, 'FractionArea', @fraction_area)
-      add_element_in_xml_file(xml_element, ns, 'SpaceTypesFloorArea', @space_types_floor_area)
-      add_element_in_xml_file(xml_element, ns, 'ConditionedFloorAreaHeatedOnly', @conditioned_floor_area_heated_only)
-      add_element_in_xml_file(xml_element, ns, 'ConditionedFloorAreaCooledOnly', @conditioned_floor_area_cooled_only)
-      add_element_in_xml_file(xml_element, ns, 'ConditionedFloorAreaHeatedCooled', @conditioned_floor_area_heated_cooled)
-      add_element_in_xml_file(xml_element, ns, 'ConditionedBelowGradeFloorArea', @conditioned_below_grade_floor_area)
-      add_element_in_xml_file(xml_element, ns, 'CustomConditionedAboveGradeFloorArea', @custom_conditioned_above_grade_floor_area)
-      add_element_in_xml_file(xml_element, ns, 'CustomConditionedBelowGradeFloorArea', @custom_conditioned_below_grade_floor_area)
+      add_user_defined_field_to_xml_file(xml_element, ns, 'TotalFloorArea', @total_floor_area)
```

Comment:
Is `TotalFloorArea` the same as the gross floor area?  If so, this should not be added as a UDF.  Same goes for the other floor areas below:
```xml

              <auc:FloorAreas>
                <auc:FloorArea>
                  <auc:FloorAreaType>Cooled only</auc:FloorAreaType>
                  <auc:FloorAreaValue>0.0</auc:FloorAreaValue>
                </auc:FloorArea>
                <auc:FloorArea>
                  <auc:FloorAreaType>Gross</auc:FloorAreaType>
                  <auc:FloorAreaValue>49390</auc:FloorAreaValue>
                </auc:FloorArea>
                <auc:FloorArea>
                  <auc:FloorAreaType>Heated and Cooled</auc:FloorAreaType>
                  <auc:FloorAreaValue>49261</auc:FloorAreaValue>
                </auc:FloorArea>
                <auc:FloorArea>
                  <auc:FloorAreaType>Heated only</auc:FloorAreaType>
                  <auc:FloorAreaValue>129</auc:FloorAreaValue>
                </auc:FloorArea>
              </auc:FloorAreas>
```

Reply:
@mrigeshgupta please look at: https://github.com/BuildingSync/BuildingSync-gem/blob/7117b72a443496d9d74e93461b597e85664f3bbe/lib/buildingsync/model_articulation/spatial_element.rb#L54 to write the properties to the same XML elements as we read them from. 
We want to write those only if they are not null.